util/collate/collate.go

```diff
@@ -90,6 +90,26 @@ func NewCollationEnabled() bool {
 	return newCollationEnabled
 }
 
+// RewriteNewCollationIDIfNeeded rewrites a collation id if the new collations are enabled.
+// When new collations are enabled, we turn the collation id to negative so that other the
+// components of the cluster(for example, TiKV) is able to aware of it without any change to
+// the protocol definition.
+// When new collations are not enabled, collation id remains the same.
+func RewriteNewCollationIDIfNeeded(id int32) int32 {
+	if newCollationEnabled && id > 0 {
```

Comment:
`id > 0` can be removed?

Reply:
This is a guarantee for unexpected situation. I think we can keep it.

Reply:
I've added a warning here.